c++/src/ColumnWriter.cc

```diff
@@ -130,6 +130,8 @@ namespace orc {
         hasNullValue = true;
       }
     }
+
+    batch.decodeDictionary();
```

Comment:
My intention to add `decodeDictionary()` is that caller should call `decodeDictionary()` before calling `ColumnWriter::add()`. This can make the implementation of `ColumnWriter ` simpler.